### ISSUE TITLE
Sidebar Upsell Nudge in WP Admin

### ISSUE THREAD

**Issue by tjcafferkey** [Automattic/wp-calypso#52368]
**What**
Show upsell nudges in in WP Admin sidebar when Nav Unification is enabled so it is consistent with Calypso.

![116316298-9730e980-a766-11eb-9c7f-13a73addccec](https://user-images.githubusercontent.com/8639742/116374167-f3971600-a805-11eb-85da-93a650ab65e7.png)

**Why**
Upsell nudges are a source of revenue and since the launch of Nav Unification more users are exposed to WP Admin where these components are not shown. More information here pbAPfg-1xC-p2

**How**
This could potentially be handled in the same way the site card is rendered in WP Admin.

### Kind of nudges
Upsell Nudges in _sidebar_ might be **one** of the following:
1. JITM messages from WPCOM (`free_to_paid_plan`, `domain_to_paid_plan`, `blogger_to_personal_plan`, `wpcom_flash_sale_promotion`). The JITM architecture does a good work at returning the most suitable nudge for the current site.
https://github.com/Automattic/wp-calypso/blob/4265ac7786b2331f5c89656650f9b3bd7e3020a6/client/my-sites/current-site/notice.jsx#L276-L283
eg ![](https://cln.sh/MhxEvu+)
2. `domainCreditNotice`. Actually this isn't an **Upsell** nudge, meaning that there is no direct revenue from a User adding a domain to their site. Probably though, increases the engagement of the User with the platform. 
https://github.com/Automattic/wp-calypso/blob/4265ac7786b2331f5c89656650f9b3bd7e3020a6/client/my-sites/current-site/notice.jsx#L285
eg ![](https://cln.sh/RP4GRm+)
3. if none of the above appears, then a `domain-upsell-nudge` might appear. The main condition is that you already have bought a Domain before 7 days.
https://github.com/Automattic/wp-calypso/blob/4265ac7786b2331f5c89656650f9b3bd7e3020a6/client/my-sites/current-site/notice.jsx#L286
eg ![](https://cln.sh/7EwXYd+)

### Facts:
- both (1) & (3) drive revenue, we should focus first on displaying them in WP Admin sidebar. 
- only one nudge can be displayed at a time.
- JITM _may_ directly serve any of the (3) nudges (needs implementation for (2) & (3)).
- I have already created a POC D60968-code using JITM for (1) for Simple sites.

### Suggested technical approach
Replicating the logic for nudges (2) & (3) and having the code both in Calypso and WPCOM doesn't sound like a good choice. So I am suggesting a stepped approach:

- [x] Add JITM nudges (1) in Simple sites https://github.com/Automattic/jetpack/pull/19773
- [x] Add JITM nudges (1) in Atomic sites https://github.com/Automattic/jetpack/pull/19773
- [x] Add `domain-upsell-nudge` (3) logic in WPCOM `jitm-engine.php` so that it can be served through JITM and remove the code from Calypso
- [x] Add `domainCreditNotice` (2) logic in WPCOM `jitm-engine.php` so that it can be served through JITM and remove the code from Calypso
- [ ] Add nudges for Atomic Free sites in `jitm-engine.php`

Related to https://github.com/Automattic/wp-calypso/issues/45574

**Comment by tjcafferkey** [comment]
De-prioritising this due to pbAPfg-1xC-p2#comment-3035

**Comment by cpapazoglou** [comment]
I created a POC of `free to paid` upsell nudges being displayed to WP Admin of Simple sites D60968-code

![](https://cln.sh/UXlY4b+)

**Comment by cpapazoglou** [comment]
After https://github.com/Automattic/jetpack/pull/19773, only the `domain-upsell-nudge` and `domainCreditNotice` will be missing from WP Admin. I am not sure though, whether there is enough value in adding them currently. 

@tjcafferkey any suggestions here?

**Comment by tjcafferkey** [comment]
@cpapazoglou let's leave them until we know more about the value in adding them and leave this issue open.

**Comment by dsas** [comment]
Working on 3 `domain-upsell-nudge`, I've pushed the jitm infra changes to D69554-code and am working on the jitm itself.

**Comment by dsas** [comment]
For 3 `domain-upsell-nudge` I now have three phab diffs to review: D69689-code, D69554-code & D69690-code, I've created a separate card to track that on the [nav unification board](https://github.com/orgs/Automattic/projects/311/views/8).

**Comment by dsas** [comment]
### domain-upsell-nudge diffs

- [x] 🚢 D69787-code - mark existing sidebar jitms as undismissable 
- [x] 🚢  D69689-code - refactor the product list to be accessible from serverside code
- [x] 🚢 D69554-code - add registered_domain_within scoring rule
- [x] 🚢  #57832 - calypso - make sidebar jitms dismissable
- [x] 🚢  [21883 - jetpack](https://github.com/Automattic/jetpack/pull/21883) - make sidebar jitms dismissable
- [x] ⛔  - ~~calypso - correct jitm logic so we don't see double~~ Just going to remove the code from Calypso (see below) rather than trying to fix #58613, see D69690-code
- [x]  🚢  #58652 - remove domain-upsell-nudge and domainCreditNotice logic from Calypso
 - [x] 🚢   D69690-code - add jitm message for upselling domains and another message for using domain credit - blocked by previous items

**Comment by dsas** [comment]
From the [opening comment](https://github.com/Automattic/wp-calypso/issues/52368#issue-869695532)
>  Add nudges for Atomic Free sites in jitm-engine.php

Does anyone have any idea what this is referencing?

**Comment by mmtr** [comment]
@dsas I guess it refers to support Atomic sites without any paid plan (something that we're preventing after our work on the Atomic Downgrade pod, but we still have a large backlog of Atomic sites without a plan).

In theory, JITMs should just work for those sites, but we may have an edge case there.

**Comment by dsas** [comment]
I've just double checked and they do already work @mmtr. They should work in theory too - as part of Atomic Downgrades we added a JITM!

I'll close this issue now then.
